ruby_on_rails/testing/unit_testing.md

```diff
@@ -0,0 +1,74 @@
+### Introduction
+
+You may remember testing from the Ruby section and the emphasis on the importance of testing. Similar to that section, testing in Ruby on Rails is important to ensure your application is stable and to catch bugs before you expand your application even more! The first kind of testing that we will cover will be unit tests.
+
+### Lesson Overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+*   What are unit tests?
+*   Why are unit tests important?
+*   Unit tests in the context of Ruby on Rails.
+
+## What are unit tests?
+
+A unit test is testing a single unit of code to ensure that method/unit works as it is supposed to! In Ruby on Rails your models are one of the areas that should be covered with unit tests. These model tests specifically are used to test the logic behind your model objects. This can include validating user inputs from form submissions and other logic between models. Such as if you need to ensure your object method for generating discounts work properly. It's important as you scale your projects to ensure that the logic behind your models works as your project consistently expands. This helps make the project more maintainable as the tests provide a safety net to ensure existing functionality continues working as expected when models need to be expanded or modified. 
+
+Earlier you used RSpec for testing projects in the Ruby section. While Ruby on Rails comes with it's own testing framework. RSpec is a very popular framework for testing even in Ruby on Rail's. As such we will be using RSpec for the Ruby on Rails testing section. 
+
+## Setting up RSpec
+
+Installing RSpec for rails is a relatively straightforward task.
+
+First we add it the development and test group of the gemfile. Navigate to the gemfile and add this to the development/test section.
+
+~~~
+
+group :development, :test do
+  gem 'rspec-rails'
+end
+
+~~~
+
+Next visit the [rails installation guide](https://github.com/rspec/rspec-rails#installation) to finish the rest of the steps. Afterwards you will see an additional spec folder on the top level of your project.
```

Comment:
Nit: Having "rails installation guide" as the link makes it look like this has to do with rails, not rspec. 
```suggestion
Next visit the [rspec-rails installation guide](https://github.com/rspec/rspec-rails#installation) to finish the rest of the steps. Afterwards you will see an additional spec folder on the top level of your project.
```